strongbox-data-service/src/main/java/org/carlspring/strongbox/gremlin/dsl/EntityTraversalDsl.java

```diff
@@ -112,4 +115,24 @@ public String toString()
                                                                              ((Element) t.get()).property("uuid")
                                                                                                 .orElse("null"))));
     }
+    
+    default <E> GraphTraversal<S, E> property(final String key,
+                                              final Set<String> values,
+                                              GraphTraversal<S, E> graphTraversal)
+    {
+
+        if (!CollectionUtils.isEmpty(values))
+        {
+            if (values.size() == 1)
```

Comment:
@anki2189 why this condition needed?

Reply:
This condition is added to handle single element Set use case, which will not work if we will not add empty string.

Reply:
I think we can omit this condition and always add empty element here.

Reply:
Changed.